### ISSUE TITLE
Clearing input value using select all and backspace doesn't fire input event

### ISSUE THREAD

**Issue by Yokhtop** [JsDaddy/ngx-mask#287]
Hi. Assigning a prefix as input value in `keydown` event when the prefix is empty prevents firing `input` event.

```
@HostListener('keydown', ['$event'])
  public a(e: KeyboardEvent): void {
    const el: HTMLInputElement = e.target as HTMLInputElement;
    if (e.keyCode === 38) {
      e.preventDefault();
    }
    if (e.keyCode === 37 || e.keyCode === 8) {
      if ((el.selectionStart as number) <= this._maskService.prefix.length
        && (el.selectionEnd as number) <= this._maskService.prefix.length) {
        e.preventDefault();
      }
      this.onFocus(e);
      if (e.keyCode === 8
        && el.selectionStart === 0
        && el.selectionEnd === el.value.length) {
        el.value = this._maskService.prefix;
        this._position = this._maskService.prefix ? this._maskService.prefix.length : 1;
        this.onInput(e);
      }
    }
  }
```

Thanks.

**Comment by gedvan** [comment]
The `change` event is not been triggered as well.

**Comment by Yokhtop** [comment]
I guess it can be fixed by wrapping the line:
`el.value = this._maskService.prefix;`
Into if statement:
```
if (this._maskService.prefix.length) {
  el.value = this._maskService.prefix;
}
```

**Comment by dzonatan** [comment]
@Thegrep01 why this issue was closed?
I don't see this to be fixed in `7.3.6`.

**Comment by Yokhtop** [comment]
@Thegrep01 I can't reproduce this in `7.6.4`
https://stackblitz.com/edit/angular-ztu37x
